### ISSUE TITLE
borg mount throws "Transport endpoint is not connected"

### ISSUE THREAD

**Issue by tongpu** [borgbackup/borg#4534]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

#### Your borg version (borg -V).
1.1.8

#### Operating system (distribution) and version.
Arch Linux (rolling release, freshly updated)

#### Hardware / network configuration, and filesystems used.
Filesystem client: ext4
Filesystem server: zfs

#### How much data is handled by borg?
```
                       Original size      Compressed size    Deduplicated size
All archives:              422.73 GB            117.54 GB             10.41 GB

                       Unique chunks         Total chunks
Chunk index:                  216213              3273861
```

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
`borg mount :: /mnt/borg borg -f --debug`

## Describe the problem you're observing.
```
$ cd /mnt/borg
$ ls
example-2019-04-11T09:15:34.728757/
$ cd example-2019-04-25T09:16:27.586479/
$ ls
ls: cannot open directory '.': Transport endpoint is not connected
```

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
Reproducible 100% of the time.

#### Include any warning/errors/backtraces from the system logs
```
borg mount :: /mnt/borg borg -f --debug
using builtin fallback logging configuration
35 self tests completed in 0.09 seconds
SSH command line: ['ssh', 'user@192.168.1.1', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.28 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/~/local-user/ein'
Remote: Resolved repository path to '/home/backup/user/borg'
Remote: Verified integrity of /home/backup/user/borg/index.183
TAM-verified manifest
security: read previous location 'ssh://user@192.168.1.1/~/local-user/borg'
security: read manifest timestamp '2019-04-25T07:17:48.492891'
security: determined newest manifest timestamp as 2019-04-25T07:17:48.492891
security: repository checks ok, allowing access
mount data cache capacity: 4 chunks
Mounting filesystem
Initializing llfuse
Calling fuse_mount
Calling fuse_lowlevel_new
Calling fuse_session_add_chan
Calling fuse_session_loop
handler raised <class 'TypeError'> exception (unpack() takes no arguments (1 given)), terminating main loop.
Terminated main loop because request handler raised exception, re-raising..
Calling fuse_session_remove_chan
Calling fuse_session_destroy
RepositoryCache: current items 1, size 18.14 kB / 2.15 GB, 0 hits, 1 misses, 0 slow misses (+0.0s), 0 evictions, 0 ENOSPC hit
RemoteRepository: 5.37 kB bytes sent, 2.98 MB bytes received, 107 messages sent
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 4436, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 4368, in run
    return set_ec(func(args))
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 1353, in do_mount
    return self._do_mount(args)
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 152, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 1363, in _do_mount
    operations.mount(args.mountpoint, args.options, args.foreground)
  File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 340, in mount
    signal = fuse_main()
  File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 34, in fuse_main
    return llfuse.main(workers=1)
  File "src/fuse_api.pxi", line 330, in llfuse.main
  File "src/handlers.pxi", line 39, in llfuse.fuse_lookup
  File "src/handlers.pxi", line 40, in llfuse.fuse_lookup
  File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 557, in lookup
    self._load_pending_archive(parent_inode)
  File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 554, in _load_pending_archive
    self.process_archive(archive_name, [os.fsencode(archive_name)])
  File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 377, in process_archive
    consider_part_files=self.args.consider_part_files):
  File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 160, in iter_archive_items
    item = unpacker.unpack(write_bytes)
TypeError: unpack() takes no arguments (1 given)

Platform: Linux example 5.0.9-arch1-1-ARCH #1 SMP PREEMPT Sat Apr 20 15:00:46 UTC 2019 x86_64
Linux: arch  
Borg: 1.1.8  Python: CPython 3.7.3
PID: 8011  CWD: /home/local-user
sys.argv: ['/usr/bin/borg', 'mount', '::', '/mnt/borg', 'borg', '-f', '--debug']
SSH_ORIGINAL_COMMAND: None
```

**Comment by ThomasWaldmann** [comment]
You run borg with an unsupported msgpack version.

This is a known archlinux issue.

You can uninstall the arch borg package and use our binary from github releases page.

On April 29, 2019 1:57:28 PM GMT+02:00, Lukas Grossar <notifications@github.com> wrote:
>## Have you checked borgbackup docs, FAQ, and open Github issues?
>
>Yes
>
>## Is this a BUG / ISSUE report or a QUESTION?
>
>ISSUE
>
>## System information. For client/server mode post info for both
>machines.
>
>#### Your borg version (borg -V).
>1.1.8
>
>#### Operating system (distribution) and version.
>Arch Linux (rolling release, freshly updated)
>
>#### Hardware / network configuration, and filesystems used.
>Filesystem client: ext4
>Filesystem server: zfs
>
>#### How much data is handled by borg?
>```
>                Original size      Compressed size    Deduplicated size
>All archives:              422.73 GB            117.54 GB            
>10.41 GB
>
>                       Unique chunks         Total chunks
>Chunk index:                  216213              3273861
>```
>
>#### Full borg commandline that lead to the problem (leave away
>excludes and passwords)
>`borg mount :: /mnt/borg borg -f --debug`
>
>## Describe the problem you're observing.
>```
>$ cd /mnt/borg
>$ ls
>example-2019-04-11T09:15:34.728757/
>$ cd example-2019-04-25T09:16:27.586479/
>$ ls
>ls: cannot open directory '.': Transport endpoint is not connected
>```
>
>#### Can you reproduce the problem? If so, describe how. If not,
>describe troubleshooting steps you took before opening the issue.
>Reproducible 100% of the time.
>
>#### Include any warning/errors/backtraces from the system logs
>```
>borg mount :: /mnt/borg borg -f --debug
>using builtin fallback logging configuration
>35 self tests completed in 0.09 seconds
>SSH command line: ['ssh', 'user@192.168.1.1', 'borg', 'serve',
>'--umask=077', '--debug']
>Remote: using builtin fallback logging configuration
>Remote: 35 self tests completed in 0.28 seconds
>Remote: using builtin fallback logging configuration
>Remote: Initialized logging system for JSON-based protocol
>Remote: Resolving repository path b'/~/local-user/ein'
>Remote: Resolved repository path to '/home/backup/user/borg'
>Remote: Verified integrity of /home/backup/user/borg/index.183
>TAM-verified manifest
>security: read previous location
>'ssh://user@192.168.1.1/~/local-user/borg'
>security: read manifest timestamp '2019-04-25T07:17:48.492891'
>security: determined newest manifest timestamp as
>2019-04-25T07:17:48.492891
>security: repository checks ok, allowing access
>mount data cache capacity: 4 chunks
>Mounting filesystem
>Initializing llfuse
>Calling fuse_mount
>Calling fuse_lowlevel_new
>Calling fuse_session_add_chan
>Calling fuse_session_loop
>handler raised <class 'TypeError'> exception (unpack() takes no
>arguments (1 given)), terminating main loop.
>Terminated main loop because request handler raised exception,
>re-raising..
>Calling fuse_session_remove_chan
>Calling fuse_session_destroy
>RepositoryCache: current items 1, size 18.14 kB / 2.15 GB, 0 hits, 1
>misses, 0 slow misses (+0.0s), 0 evictions, 0 ENOSPC hit
>RemoteRepository: 5.37 kB bytes sent, 2.98 MB bytes received, 107
>messages sent
>Local Exception
>Traceback (most recent call last):
>File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 4436, in
>main
>    exit_code = archiver.run(args)
>File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 4368, in
>run
>    return set_ec(func(args))
>File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 1353, in
>do_mount
>    return self._do_mount(args)
>File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 152, in
>wrapper
>    return method(self, args, repository=repository, **kwargs)
>File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 1363, in
>_do_mount
>    operations.mount(args.mountpoint, args.options, args.foreground)
>File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 340, in
>mount
>    signal = fuse_main()
>File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 34, in
>fuse_main
>    return llfuse.main(workers=1)
>  File "src/fuse_api.pxi", line 330, in llfuse.main
>  File "src/handlers.pxi", line 39, in llfuse.fuse_lookup
>  File "src/handlers.pxi", line 40, in llfuse.fuse_lookup
>File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 557, in
>lookup
>    self._load_pending_archive(parent_inode)
>File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 554, in
>_load_pending_archive
>    self.process_archive(archive_name, [os.fsencode(archive_name)])
>File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 377, in
>process_archive
>    consider_part_files=self.args.consider_part_files):
>File "/usr/lib/python3.7/site-packages/borg/fuse.py", line 160, in
>iter_archive_items
>    item = unpacker.unpack(write_bytes)
>TypeError: unpack() takes no arguments (1 given)
>
>Platform: Linux example 5.0.9-arch1-1-ARCH #1 SMP PREEMPT Sat Apr 20
>15:00:46 UTC 2019 x86_64
>Linux: arch  
>Borg: 1.1.8  Python: CPython 3.7.3
>PID: 8011  CWD: /home/local-user
>sys.argv: ['/usr/bin/borg', 'mount', '::', '/mnt/borg', 'borg', '-f',
>'--debug']
>SSH_ORIGINAL_COMMAND: None
>```
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/borgbackup/borg/issues/4534

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by tongpu** [comment]
Sorry, I was only looking in the open issues. Do you think it would help to keep an issue open for that error?

**Comment by ThomasWaldmann** [comment]
Good idea.

So reopening this as "wontfix" so people can better find it.

**Comment by Fryboyter** [comment]
> You can uninstall the arch borg package and use our binary from github releases page.

For Arch users, borg-bin (https://aur.archlinux.org/packages/borg-bin) is now available in the AUR. Only as a small hint. :-)

**Comment by ThomasWaldmann** [comment]
There is some work going on in #4560 which might resolve the issue by assimi.... ehrm bundling a compatible msgpack version into borg.

**Comment by ThomasWaldmann** [comment]
TODO: close this after everybody has >= 1.1.10 (see above).

**Comment by tongpu** [comment]
borg 1.1.10 has been updated in the community repo since 2019-05-25, so I would believe it's safe to close this issue.

**Comment by ThomasWaldmann** [comment]
Yeah, but this also affected other dists than arch linux, so keeping this open for a while might avoid duplicate bug reports.

**Comment by markuspachali** [comment]
I have an issue with looks very similar to this one, so before I open a new issue I'll try it with this one.

If I try to mount my backup I got this (similar as initial requestor):
> [root@test01 ~]# cd /mnt/backup
-bash: cd: /mnt/backup: Transport endpoint is not connected
[root@test01 ~]#

Debug output is this:
>[root@test01 ~]# borg mount borg-backup@infrastructure01.example.com:/home/borg-backup/repositories/test01.example.com /mnt/backup -f --debug
using builtin fallback logging configuration
35 self tests completed in 0.21 seconds
SSH command line: ['ssh', '-i', '/root/.ssh/borg-backup_id_rsa', 'borg-backup@infrastructure01.example.com', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.20 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/home/borg-backup/repositories/test01.example.com'
Remote: Resolved repository path to '/home/borg-backup/repositories/test01.example.com'
Remote: Verified integrity of /home/borg-backup/repositories/test01.example.com/index.5
TAM-verified manifest
security: read previous location 'ssh://borg-backup@infrastructure01.example.com/home/borg-backup/repositories/test01.example.com'
security: read manifest timestamp '2020-04-10T20:12:21.183029'
security: determined newest manifest timestamp as 2020-04-10T20:12:21.183029
security: repository checks ok, allowing access
mount data cache capacity: 4 chunks
Mounting filesystem
Initializing llfuse
Calling fuse_mount
Calling fuse_lowlevel_new
Calling fuse_session_add_chan
Calling fuse_session_loop
handler raised <class 'AttributeError'> exception ('llfuse.EntryAttributes' object has no attribute 'st_birthtime_ns'), terminating main loop.
Terminated main loop because request handler raised exception, re-raising..
Calling fuse_session_remove_chan
Calling fuse_session_destroy
RepositoryCache: current items 0, size 0 B / 1.44 GB, 0 hits, 0 misses, 0 slow misses (+0.0s), 0 evictions, 0 ENOSPC hit
RemoteRepository: 257 B bytes sent, 1.67 kB bytes received, 4 messages sent
Local Exception
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/borg/archiver.py", line 4529, in main
    exit_code = archiver.run(args)
  File "/usr/lib64/python3.6/site-packages/borg/archiver.py", line 4461, in run
    return set_ec(func(args))
  File "/usr/lib64/python3.6/site-packages/borg/archiver.py", line 1387, in do_mount
    return self._do_mount(args)
  File "/usr/lib64/python3.6/site-packages/borg/archiver.py", line 166, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib64/python3.6/site-packages/borg/archiver.py", line 1397, in _do_mount
    operations.mount(args.mountpoint, args.options, args.foreground)
  File "/usr/lib64/python3.6/site-packages/borg/fuse.py", line 351, in mount
    signal = fuse_main()
  File "/usr/lib64/python3.6/site-packages/borg/fuse.py", line 34, in fuse_main
    return llfuse.main(workers=1)
  File "src/fuse_api.pxi", line 343, in llfuse.main (src/llfuse.c:35968)
  File "src/handlers.pxi", line 80, in llfuse.fuse_getattr (src/llfuse.c:8699)
  File "src/handlers.pxi", line 81, in llfuse.fuse_getattr (src/llfuse.c:8646)
  File "/usr/lib64/python3.6/site-packages/borg/fuse.py", line 544, in getattr
    entry.st_birthtime_ns = item.get('birthtime', mtime_ns)
AttributeError: 'llfuse.EntryAttributes' object has no attribute 'st_birthtime_ns'

>Platform: Linux test01 5.3.18-2-pve #1 SMP PVE 5.3.18-2 (Sat, 15 Feb 2020 15:11:52 +0100) x86_64
Linux: CentOS Linux 7.7.1908 Core
Borg: 1.1.11  Python: CPython 3.6.8 msgpack: 0.5.6
PID: 11603  CWD: /root
sys.argv: ['/usr/bin/borg', 'mount', 'borg-backup@infrastructure01.example.com:/home/borg-backup/repositories/test01.example.com', '/mnt/backup', '-f', '--debug']
SSH_ORIGINAL_COMMAND: None

Hope someone has an idea how to get this working :)

**Comment by ThomasWaldmann** [comment]
@markuspachali this is already fixed in 1.1-maint branch and will be in 1.1.12 soon. It is caused by an accidentally raised llfuse version requirement (>= 1.3.0). You could try the fat binary from github releases for a quick remedy, or install a newer llfuse or wait for 1.1.12 (which will work again with older llfuse).

**Comment by ThomasWaldmann** [comment]
as a general note: "transport endpoint is not connected" usually just means the `borg mount` process has died due to some uncaught exception crashing it. that can have all sorts of root causes.

to find the real root cause, the usual method is to `borg mount -f ...` and then look at the traceback when it crashes.

as it does not make sense to collect them all in this ticket, I am closing and locking it now.
